src/app/EventLogging.h

```diff
@@ -61,7 +63,46 @@ class EventLogger : public EventLoggingDelegate
  *
  * @return CHIP_ERROR  CHIP Error Code
  */
-template <typename T>
+template <typename T,
+          typename std::enable_if_t<std::is_class<T>::value && DataModel::IsFabricScopedEvent<T>::value &&
+                                        std::is_same<decltype(std::declval<T>().GetFabricIndex()), FabricIndex>::value,
+                                    T> * = nullptr>
+CHIP_ERROR LogEvent(const T & aEventData, EndpointId aEndpoint, EventNumber & aEventNumber,
+                    EventOptions::Type aUrgent = EventOptions::Type::kNotUrgent)
+{
+    EventLogger<T> eventData(aEventData);
+    ConcreteEventPath path(aEndpoint, aEventData.GetClusterId(), aEventData.GetEventId());
+    EventManagement & logMgmt = chip::app::EventManagement::GetInstance();
+    EventOptions eventOptions;
+    eventOptions.mUrgent   = aUrgent;
+    eventOptions.mPath     = path;
+    eventOptions.mPriority = aEventData.GetPriorityLevel();
+    eventOptions.mFabricIndex.SetNonNull(aEventData.GetFabricIndex());
+    ChipLogError(EventLogging, "current fabric index is %u in LogEvent", aEventData.GetFabricIndex());
+    ;
+    return logMgmt.LogEvent(&eventData, eventOptions, aEventNumber);
+}
+
+template <typename T,
+          typename std::enable_if_t<
+              std::is_class<T>::value && DataModel::IsFabricScopedEvent<T>::value &&
+                  std::is_same<decltype(std::declval<T>().GetFabricIndex()), DataModel::Nullable<chip::FabricIndex>>::value,
+              T> * = nullptr>
+CHIP_ERROR LogEvent(const T & aEventData, EndpointId aEndpoint, EventNumber & aEventNumber,
+                    EventOptions::Type aUrgent = EventOptions::Type::kNotUrgent)
+{
+    EventLogger<T> eventData(aEventData);
+    ConcreteEventPath path(aEndpoint, aEventData.GetClusterId(), aEventData.GetEventId());
+    EventManagement & logMgmt = chip::app::EventManagement::GetInstance();
+    EventOptions eventOptions;
+    eventOptions.mUrgent      = aUrgent;
+    eventOptions.mPath        = path;
+    eventOptions.mPriority    = aEventData.GetPriorityLevel();
+    eventOptions.mFabricIndex = aEventData.GetNullableFabricIndex();
```

Comment:
Nit: Should be GetFabricIndex()

Reply:
Done, thanks, It is ready for formal review.